shm/map.hpp

```diff
@@ -0,0 +1,17 @@
+#pragma once
+#include <vector>
+#include <memory>
+#include "island.hpp"
+
+#define MAX_ISLANDS 10
```

Comment:
move to class. If this value is needed only in .cpp file then move to the anonymous namespace in .cpp file.